contracts/infrastructure/ens/ArgentENSManager.sol

```diff
@@ -87,34 +89,53 @@ contract ArgentENSManager is IENSManager, Owned, Managed {
     * Registers both the forward and reverse ENS.
     * @param _label The subdomain label.
     * @param _owner The owner of the subdomain.
+    * @param _managerSignature The manager signature of the hash of _owner and _label.
     */
-    function register(string calldata _label, address _owner) external onlyManager {
+    function register(string calldata _label, address _owner, bytes calldata _managerSignature) external override validateENSLabel(_label) {
+        bytes32 signedHash = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", keccak256(abi.encodePacked(_owner, _label))));
+        validateManagerSignature(signedHash, _managerSignature);
+
         bytes32 labelNode = keccak256(abi.encodePacked(_label));
         bytes32 node = keccak256(abi.encodePacked(rootNode, labelNode));
         address currentOwner = ensRegistry.owner(node);
-        require(currentOwner == address(0), "AEM: _label is alrealdy owned");
+        require(currentOwner == address(0), "AEM: label is already owned");
 
         // Forward ENS
         ensRegistry.setSubnodeRecord(rootNode, labelNode, _owner, address(ensResolver), 0);
         ensResolver.setAddr(node, _owner);
 
-        // Reverse ENS
-        strings.slice[] memory parts = new strings.slice[](2);
-        parts[0] = _label.toSlice();
-        parts[1] = rootName.toSlice();
-        string memory name = ".".toSlice().join(parts);
-        ENSReverseRegistrar reverseRegistrar = ENSReverseRegistrar(_getENSReverseRegistrar());
-        bytes32 reverseNode = reverseRegistrar.node(_owner);
-        ensResolver.setName(reverseNode, name);
+        string memory name = string(abi.encodePacked(_label, ".", rootName));
+
+        // Optionally set the reverse ENS
+        bytes32 reverseNode = IReverseRegistrar(_getENSReverseRegistrar()).node(_owner);
+
+        if(ensRegistry.owner(reverseNode) == address(this)) {
```

Comment:
I believe this condition should be `if(ensRegistry.resolver(reverseNode) == ensResolver)` because we should only update the resolver if it has been set for the reverse record with `claimWithResolver` .